### ISSUE TITLE
Posts List: cannot search old posts

### ISSUE THREAD

**Issue by designsimply** [wordpress-mobile/WordPress-iOS#12065]
A user reported being unable to scroll back to or search old posts. Their site is a WordPress.com site and they are using iOS App version 12.5.1 on both iPad and iPhone. User is able to search and backscroll without issue when using a web browser. A few people have tried testing and have been unable to reproduce the problem on separate test sites so far.

@aerych noted the following:

<blockquote>I also tested searching for “kann der”. If I had loaded all the posts into the list then attempted a search I was able to view the post. However, if I pulled to refresh (which clears out cached posts) then searched I did not get any results. <strong>This seems like a glitch where we might be fetching searched posts from the server but only showing them if they’d been previously cached.</strong> I don’t recall exactly how the search logic works but I’d guess either we’re saving searched posts into a different context (maybe?) or otherwise filtered in a way that we aren’t picking them up out of core data.</blockquote>

I tried that for my own test blog and was unable to replicate the exact same problem on a test site.

Steps to reproduce:

1. Start with a site that has enough posts to use infinite scroll on the posts list screen.
1. Try searching for various types of posts.
1. Scroll forward in the post list until you find a sample post to search for in the backscroll.
1. Scroll back up and refresh the post list view.
1. Try searching for the post found in step 3.
1. Try toggling the me/everyone filter and search again.

<strong>Result:</strong> in my own testing I could always find the posts when searching for a word or a few words from the title—I couldn't see posts when I tried searching for words in the post body and filed a separate issue for that at https://github.com/wordpress-mobile/WordPress-iOS/issues/12064.

<img src="https://user-images.githubusercontent.com/1119271/60631850-f74ee280-9dbe-11e9-86da-503849ce6ff5.PNG" width="270" alt="IMG_2877"> <img src="https://user-images.githubusercontent.com/1119271/60631853-f918a600-9dbe-11e9-8e9f-93321827a825.PNG" width="270" alt="IMG_2876">

<sup>Tested with WP Internal 12.8.0.20190701 on iPhone 6S iOS 12.3.1.</sup>

This type of snippet from the app logs was referenced in the mobile requests but in my own blog I can see the same types of logs but can still find those posts in the search results:

```
2019-07-03 17:27:38:879 Deleting Post: <Post: 0x280aa0e70> (entity: Post; id: 0xe8c0ff3bb2d6f333 <x-coredata://36751460-704F-4FBB-8497-BEF733573DFD/Post/p274> ; data: {
    author = Sheri;
    authorAvatarURL = "https://1.gravatar.com/avatar/198723e26f9350d9bbe8d4f35a8b0bb7?s=96&d=identicon&r=G";
    authorID = 396702;
    blog = "0xe8c0ff3baa66f319 <x-coredata://36751460-704F-4FBB-8497-BEF733573DFD/Blog/p1854>";
    categories =    (
        "0xe8c0ff3bb23ef305 <x-coredata://36751460-704F-4FBB-8497-BEF733573DFD/Category/p296>"
    );
    commentCount = 0;
    comments =    (
    );
    content = "Steamed lungo milk, shop percolator, foam cream grounds body cream. So aromatic flavour caffeine, wings robust black rich et milk. Mocha affogato, irish, wings siphon sweet blue mountain chicory milk ";
    dateModified = "2016-07-27 22:32:36 +0000";
    "date_created_gmt" = "2016-07-27 21:31:01 +0000";
    disabledPublicizeConnections = "{\n}";
    featuredImage = nil;
    geolocation = nil;
    isStickyPost = 0;
    latitudeID = nil;
    likeCount = 0;
    longitudeID = nil;
    media =    (
    );
    metaIsLocal = 0;
    metaPublishImmediately = 0;
    "mt_excerpt" = "";
    original = nil;
    password = nil;
    pathForDisplayImage = "";
    permaLink = "https://design5279.wordpress.com/2016/07/28/coffee-ipsum/";
    postFormat = standard;
    postID = 10161;
    postTitle = "Coffee Ipsum";
    postType = post;
    publicID = nil;
    publicizeMessage = nil;
    publicizeMessageID = nil;
    remoteStatusNumber = 3;
    revision = nil;
    revisions = "(\n    10162\n)";
    status = publish;
    "suggested_slug" = nil;
    tags = "coffee,coffee ipsum,ipsum";
    "wp_slug" = "coffee-ipsum";
})
```

<sub>(internal references: p4a5px-2pn-p2, 2096224-zen)</sub>

**Comment by rezzap** [comment]
I have another report of this issue by a WordPress.com user in 3161501-zen

They have a total of 317 posts on their site and using the title of the post to search does not return any results for some of the older posts. The two posts, in particular, they are searching from were published in 2017. 

I was able to replicate this as the user on iOS app 15.3. 

I recalled this was also an issue in Android and was fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/11098. If that's any help to getting this working in iOS.
